doc/hotspot-style.md

```diff
@@ -651,6 +651,21 @@ constant members.  Compilers having such bugs are no longer supported.
 Except where an enum is semantically appropriate, new code should use
 integral constants.
 
+### alignas
+
+`alignas`
+([n1877](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2005/n1877.pdf))
+is permitted for the purpose of imposing alignment restrictions on types or
+objects in HotSpot code. The exception is if `alignas` specifies extended
+alignment, in other words when alignment is greater than alignof(std::max_align_t)
+([C++14 3.11/3](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf)),
+for which it is only permitted for variables with static
+([C++14 3.7.1](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf))
+or automatic storage duration. It is also forbidden to have over-aligned types
+within HotSpot, at least until we switch to using C++17. See the review for
+[JDK-8252584](https://github.com/openjdk/jdk/pull/11315)
+for more information.
```

Comment:
There have been a couple of relevant defect reports:
CWG 2354 - https://cplusplus.github.io/CWG/issues/2354.html
CWG 1437 - https://cplusplus.github.io/CWG/issues/1437.html

It turns out alignas must be applied to a definition, with declarations
optionally having alignas equivalent to the definition:
C++14 10.6.2/6 - https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf.

I found the structure of the proposed text somewhat difficult to follow.

Limits on the maximum alignment need to be explored and documented.  The
maximum for an automatic variable might be significantly limited, or maybe
just risks blowing the stack.  I think we don't have any existing uses of
aligned automatic variables, so maybe we don't need to permit them?  For now
I'll leave that in place, but I could easily be convinced to remove it.

So I suggest something like the following:
https://github.com/openjdk/jdk/compare/master...kimbarrett:openjdk-jdk:alignas?expand=1